src/robotFactory.js

```diff
@@ -1,10 +1,77 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
```

Comment:
```suggestion
  constructor(
    name,
    weight,
    { 
      x = 0, 
      y = 0 
    },
    chipVersion
    ) {
```
it's better to use such formatting for better reading